test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	siteEditorStyleVariations: async (
+		{ admin, editor, page, pageUtils, requestUtils },
+		use
+	) => {
+		await use(
+			new SiteEditorStyleVariations( {
+				admin,
+				editor,
+				page,
+				pageUtils,
+				requestUtils,
+			} )
+		);
+	},
+} );
+
+test.describe( 'Global styles variations', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme(
+			'gutenberg-test-themes/style-variations'
+		);
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'should have three variations available with the first one being active', async ( {
+		admin,
+		editor,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await editor.toggleGlobalStyles();
+		await siteEditorStyleVariations.openOtherStyles();
+
+		const variations = await siteEditorStyleVariations.getAvailableStyleVariations();
+
+		expect( await variations.count() ).toEqual( 3 );
```

Comment:
Let's use [Playwright-style assertions](https://playwright.dev/docs/test-assertions#locator-assertions-to-have-count):

```suggestion
		await expect( variations ).toHaveCount( 3 );
```